go/vt/mysqlctl/backup.go

```diff
@@ -225,29 +226,32 @@ func Restore(
 	localMetadata map[string]string,
 	logger logutil.Logger,
 	deleteBeforeRestore bool,
-	dbName string) (mysql.Position, error) {
+	dbName string,
+	snapshotTime time.Time) (mysql.Position, string, error) {
```

Comment:
I thought about this and at first glance it seemed like a good idea. But we have the error cases where we return an empty position and empty string. It would be misleading to return a made-up BackupManifest when you didn't actually find a good backup.

Reply:
In the case of an error, we should return `nil` for the `*BackupManifest`.

Reply:
Duh, of course!
I have changed them to return *BackupManifest. The code is now much cleaner, thanks to your suggestion!
Even though after moving backupTime to local_metadata, the only field callers now use from BackupManifest is Position, I think it is cleaner and future-proof to continue to return *BackupManifest rather than go back to returning mysql.Position.